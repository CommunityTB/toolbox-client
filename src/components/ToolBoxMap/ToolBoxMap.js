import React, { Component } from 'react';
import { Map, GoogleApiWrapper } from 'google-maps-react';
import config from '../../config'

class ToolBoxMap extends Component {
  state = {
    showingInfoWindow: false,
    wd: '100%'
  };

  onMapReady = (mapProps, map) => {
    this.map = map;
    let mql = window.matchMedia('(max-width: 992px)');
    const screenWidthTest = (e) => {
      let wd = '100%'
      if (!e.matches) {
        /* the viewport is more than than 992 pixels wide */
        wd = '480px';
      }
      this.setState({ wd: wd })
      const currCenter = map.getCenter();
      this.props.google.maps.event.trigger(map, 'resize');
      map.setCenter(currCenter);
      
    }
    screenWidthTest(mql);
    mql.addListener(screenWidthTest);
  };

  onMarkerClick = () => this.setState({ showingInfoWindow: true });
  onInfoWindowClose = () => this.setState({ showingInfoWindow: false });

  onMapClicked = () => {
    if (this.state.showingInfoWindow)
      this.setState({ showingInfoWindow: false });
  };

  render() {
    if (!this.props.loaded) return <div>Loading...</div>;
    return (
      <div className="toolbox-map" aria-label="map">
        <Map
          initialCenter={{ lat: 38.5548665, lng: -121.6106212 }}
          className="map"
          google={this.props.google}
          onClick={this.onMapClicked}
          onReady={this.onMapReady}
          style={{ height: '360px', position: 'relative', width: this.state.wd }}
          zoom={12}
        />
      </div>
    );
  }
}

export default GoogleApiWrapper({
  apiKey: config.MAPS_API_KEY
})(ToolBoxMap);