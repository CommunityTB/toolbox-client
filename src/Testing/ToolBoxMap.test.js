import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ToolBoxMap from '../components/ToolBoxMap/ToolBoxMap'

describe(`ToolBoxMap component`, () => {
  it('renders a ToolBoxMap by default', () => {
    const wrapper = shallow(<ToolBoxMap />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})