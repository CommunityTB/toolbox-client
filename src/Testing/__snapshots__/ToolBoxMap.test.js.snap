// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBoxMap component renders a ToolBoxMap by default 1`] = `<DefaultLoadingContainer />`;
